### ISSUE TITLE
[eval] SSL support

### ISSUE THREAD

**Issue by markknol** [HaxeFoundation/haxe#7619]
I want to download a file from https link within a macro, but I get an error:

```haxe
Https is only supported with -lib hxssl
```
I installed `haxelib install hxssl` but that has errors:
```
C:\HaxeToolkit\libs\hxssl/3,0,0-alpha/sys/ssl/Socket.hx:335: characters 10-13 : Type not found : Lib
C:\HaxeToolkit\libs\hxssl/3,0,0-alpha/sys/ssl/Socket.hx:338: characters 34-37 : Type not found : Lib
```

I'm using Haxe 4.0.0-preview-5.

cc @tong : I've added you to this issue since you are original author of the hxssl library, maybe you can help?

**Comment by Simn** [comment]
I don't know if we want to have this in the compiler...

**Comment by markknol** [comment]
I would be fine to still add`-lib hxssl` or some alternative library, but it somehow has to work :)

**Comment by tong** [comment]
hxssl are neko/cpp bindings not available in macro context.
You might use something like curl with sys.io.Process here.

**Comment by ncannasse** [comment]
We definitely want to be able to make https requests in haxelib run and other eval-based tools

**Comment by tong** [comment]
> We definitely want to be able to make https requests in haxelib run and other eval-based tools

The ancient 'ssl written in haxe' issue :)

**Comment by kevinresol** [comment]
@markknol judging from the error message I guess you are using tink_http. If that's the case you can try the `CurlClient`. For `fetch()`, use this option: `{client: Curl}`.

But yes, cross platform support of the `sys.ssl.Socket` class would be nice.

**Comment by ncannasse** [comment]
We won't implement a full SSL client in Haxe, but we can use the same wrapper we use for neko/hashlink/etc. and port it to OCaml API instead.

**Comment by ncannasse** [comment]
Or instead use an available ocaml-ssl package on opam https://opam.ocaml.org/packages/ssl/

**Comment by Simn** [comment]
Linking openssl is still quite heavy...

**Comment by andyli** [comment]
There is a pure OCaml TLS implementation: https://opam.ocaml.org/packages/tls/
Not sure how easy to use, but I'm sure it's quite some work for me to package it for the Linux distros...

**Comment by ncannasse** [comment]
Could we have a more meaningful error message in the meanwhile?

**Comment by Simn** [comment]
This works now.

**Comment by markknol** [comment]
That is great news, thanks so much for this feature! 
Will it be available in Haxe 4.1?
